### PR TITLE
Open rates/notification api#221

### PR DESCRIPTION
**Pull request by saman-moshafi-va**
If GOOGLE_ANALYTICS_ENABLED equals 'True', then the celery worker will build the google analytics pixel URL and pass it to the HTMLEmaiTemplate

### CODE REVIEW

File: tests/app/delivery/test_send_to_providers.py [hunk 1]
```diff
@@ -511,18 +524,36 @@ def test_get_html_email_renderer_prepends_logo_path(notify_api):
...
+    )
+    notification = Notification(
+        id='notid',
+        subject='email subject',
+        service=service,
+        template=template
     )
 
-    renderer = send_to_providers.get_html_email_options(service)
+    renderer = send_to_providers.get_html_email_options(notification, app.govdelivery_client)
```
Comment (ffafara-tw) [inline tests/app/delivery/test_send_to_providers.py:546]: Please use a mock provider to not tie tests to any of the real providers.

File: tests/app/delivery/test_send_to_providers.py [hunk 2]
```diff
@@ -450,21 +450,27 @@ def test_send_email_should_use_service_reply_to_email(
     )
 
 
-def test_get_html_email_renderer_should_return_for_normal_service(sample_service):
-    options = send_to_providers.get_html_email_options(sample_service)
+def test_get_html_email_renderer_should_return_for_normal_service(sample_notification_model_with_organization):
+    options = send_to_providers.get_html_email_options(
+        sample_notification_model_with_organization,
+        app.govdelivery_client)
```
Comment (ffafara-tw) [inline tests/app/delivery/test_send_to_providers.py:456]: Please use a mock provider to not tie tests to any of the real providers.

File: tests/app/delivery/test_send_to_providers.py [hunk 3]
```diff
@@ -531,11 +562,26 @@ def test_get_html_email_renderer_handles_email_branding_without_logo(notify_api)
...
+    )
+    notification = Notification(
+        id='notid',
+        subject='email subject',
+        service=service,
+        template=template
     )
 
-    renderer = send_to_providers.get_html_email_options(service)
+    renderer = send_to_providers.get_html_email_options(notification, app.govdelivery_client)
```
Comment (ffafara-tw) [inline tests/app/delivery/test_send_to_providers.py:584]: Please use a mock provider to not tie tests to any of the real providers.

File: tests/app/googleanalytics/test_pixels.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
+import app.googleanalytics.pixels as gapixels
+
+import pytest
+import urllib
+
+from flask import current_app
+from app import GovdeliveryClient
+
+
+@pytest.fixture(scope='function')
```
Comment (ffafara-tw) [inline tests/app/googleanalytics/test_pixels.py:10]: Replace with a mock provider (there are already example of that in the existing tests)

File: tests/app/delivery/test_send_to_providers.py [hunk 5]
```diff
@@ -450,21 +450,27 @@ def test_send_email_should_use_service_reply_to_email(
...
+def test_get_html_email_renderer_should_return_for_normal_service(sample_notification_model_with_organization):
+    options = send_to_providers.get_html_email_options(
+        sample_notification_model_with_organization,
+        app.govdelivery_client)
     assert options['default_banner'] is True
     assert 'brand_colour' not in options.keys()
     assert 'brand_logo' not in options.keys()
     assert 'brand_text' not in options.keys()
     assert 'brand_name' not in options.keys()
+    assert 'ga_pixel_url' in options.keys()
```
Comment (ffafara-tw) [inline tests/app/delivery/test_send_to_providers.py:462]: I don't see a test with feature toggle off.

**Review by ffafara-tw — CHANGES_REQUESTED** [review]

File: app/config.py [hunk 6]
```diff
@@ -397,6 +397,12 @@ class Config(object):
     API_MESSAGE_LIMIT_ENABLED = False
     SWITCH_SLOW_SMS_PROVIDER_ENABLED = False
 
+    # Google Analytics
+
+    GOOGLE_ANALYTICS_ENABLED = str(True) == (os.getenv('GOOGLE_ANALYTICS_ENABLED', 'False'))
+    GOOGLE_ANALYTICS_URL = os.getenv('GOOGLE_ANALYTICS_URL', 'https://www.google-analytics.com/collect?')
+    GOOGLE_ANALYTICS_TID = os.getenv('GOOGLE_ANALYTICS_TID', 'UA-50123418-17')
```
[the remark this answers is not in the document]
Comment (saman-moshafi-va) [reply app/config.py:404]: non-prod, We need to update the deployment json files to define the prod value for GOOGLE_ANALYTICS_TID

File: tests/app/delivery/test_send_to_providers.py (hunk 1 above)
Comment (saman-moshafi-va) [reply tests/app/delivery/test_send_to_providers.py]: Used the mock provider defined the root conftest

File: tests/app/delivery/test_send_to_providers.py (hunk 2 above)
Comment (saman-moshafi-va) [reply tests/app/delivery/test_send_to_providers.py]: Done

File: tests/app/googleanalytics/test_pixels.py (hunk 4 above)
Comment (saman-moshafi-va) [reply tests/app/googleanalytics/test_pixels.py]: used the mock provider in the root conftest

File: tests/app/delivery/test_send_to_providers.py (hunk 5 above)
Comment (saman-moshafi-va) [reply tests/app/delivery/test_send_to_providers.py:471]: added a unit test

File: app/googleanalytics/pixels.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        'cs': provider.get_name(),
+        'cm': 'email',
+        'ci': notification.template.id
+    }
+
+    url = current_app.config['GOOGLE_ANALYTICS_URL']
+    url += '&'.join(f"{key}={value}" for key, value in url_params.items())
+
+    current_app.logger.info(
+        f"generated google analytics pixel URL is {url}"
```
[the remark this answers is not in the document]
Comment (saman-moshafi-va) [reply app/googleanalytics/pixels.py]: Done

**Pull request merged**